RayTraceAntiXray/src/main/java/com/vanillage/raytraceantixray/tasks/UpdateBukkitRunnable.java

```diff
@@ -61,14 +66,25 @@ public void run() {
                 }
 
                 BlockState blockState;
-                BlockEntity blockEntity = null;
 
                 if (result.isVisible()) {
-                    blockState = ((CraftWorld) world).getHandle().getBlockState(block);
+                    SchedulerUtil.runRegionTask(plugin, world, block.getX() >> 4, block.getZ() >> 4, () -> {
+                        BlockState blockStateLambda = ((CraftWorld) world).getHandle().getBlockState(block);
+                        sendPacketImmediately(p, new ClientboundBlockUpdatePacket(block, blockStateLambda));
 
-                    if (blockState.hasBlockEntity()) {
-                        blockEntity = ((CraftWorld) world).getHandle().getBlockEntity(block);
-                    }
+                        if (!blockStateLambda.hasBlockEntity()) {
+                            return;
+                        }
+
+                        BlockEntity blockEntity = ((CraftWorld) world).getHandle().getBlockEntity(block);
+                        Object packet = blockEntity.getUpdatePacket();
+
+                        if (packet != null) {
+                            sendPacketImmediately(p, packet);
+                        }
+
+                    });
+                    return;
```

Comment:
Not necessary with the suggestions above. Although we should probably check for `Server#isOwnedByCurrentRegion(...)` and do nothing if it's `false`. Actually we can assume that it's true because the player still has the chunk loaded.